src/api/store/community.py

```diff
@@ -1,16 +1,208 @@
+
+        zips = zipcodes.filter_by(city=city, state=state, zip_code_type='STANDARD' )
+        print("Number of zipcodes = "+str(len(zips)))
+        if len(zips)>0:
+          loc, created = Location.objects.get_or_create(location_type='CITY_ONLY', city=city, state=state)
+          if created:
+            print("City "+city+" created")
+          else:
+            print("City "+city+" found")
+
+        else:
+          print("No zipcodes found corresponding to city "+ city + ", " + state)
+          raise Exception("No zipcodes found corresponding to city "+ city + ", " + state)
+
+        self._check_geography_unique(community, geography_type, city)
+      elif geography_type == 'COUNTY':
+       # check that this county is found in the zipcodes list
+        ss = location.split('-')
+        county = ss[0]
+        if len(ss)==2:
+          state = ss[1]
+        else:
+          state = 'MA'
+
+        zips = zipcodes.filter_by(county=town, state=state, zip_code_type='STANDARD' )
+        print("Number of zipcodes = "+str(len(zips)))
+        if len(zips)>0:
+          loc, created = Location.objects.get_or_create(location_type='COUNTY_ONLY', county=county, state=state)
+        else:
+          print("No zipcodes found corresponding to county "+ county + ", " + state)
+          raise Exception("No zipcodes found corresponding to county "+ county + ", " + state)
+
+        self._check_geography_unique(community, geography_type, county)
+
+      elif geography_type == 'STATE':
+        # check that this state is found in the zipcodes list
+        state = location
+        zips = zipcodes.filter_by(state=state, zip_code_type='STANDARD' )
+        print("Number of zipcodes = "+str(len(zips)))
+        if len(zips)>0:
+          loc, created = Location.objects.get_or_create(location_type='STATE_ONLY', state=state)
+        else:
+          print("No zipcodes found corresponding to state "+ location)
+          raise Exception("No zipcodes found corresponding to state "+ location)
+
+        self._check_geography_unique(community, geography_type, location)
+
+      elif geography_type == 'COUNTRY':
+        # check that this state is found in the zipcodes list
+        country = location
+        zips = zipcodes.filter_by(country=country, zip_code_type='STANDARD' )
+        print("Number of zipcodes = "+str(len(zips)))
+        if len(zips)>0:
+          loc, created = Location.objects.get_or_create(location_type='COUNTRY_ONLY', country=country)
+        else:
+          print("No zipcodes found corresponding to country "+ location)
+          raise Exception("No zipcodes found corresponding to country "+ location)
+      
+        self._check_geography_unique(community, geography_type, location)
+
+      else:
+        raise Exception("Unexpected geography type: " + str(geography_type))
+      # should be a five character string
+      community.location_set.add(loc)
+
+  def _update_real_estate_units(self, community):
+    """
```

Comment:
Could you add some more context in the docstring on what this function does and why it is needed?

Reply:
done